src/NHibernate/Tool/hbm2ddl/SchemaMetadataUpdater.cs

```diff
@@ -29,16 +37,9 @@ public static void QuoteTableAndColumns(Configuration configuration)
 			}
 		}
 
-		private static ISet<string> GetReservedWords(IDictionary<string, string> cfgProperties)
-		{
-			var dialect = Dialect.Dialect.GetDialect(cfgProperties);
-			var connectionHelper = new ManagedProviderConnectionHelper(cfgProperties);
-			return GetReservedWords(dialect, connectionHelper);
-		}
-
 		private static ISet<string> GetReservedWords(Dialect.Dialect dialect, IConnectionHelper connectionHelper)
 		{
-			ISet<string> reservedDb = new HashSet<string>();
+			ISet<string> reservedDb = new HashSet<string>(dialect.Keywords);
```

Comment:
Causes `Update` code to try re-add all the already known words. Would be better to avoid this for `Update` case.